playwright/e2e/clusters/register-cluster.spec.ts

```diff
@@ -0,0 +1,159 @@
+    // Wait for cluster list data to load
+    await clusterListPage.waitForDataReady();
+    await clusterListPage.isClusterListScreen();
+
+    await expect(clusterListPage.registerCluster()).toBeVisible();
+    await clusterListPage.registerCluster().click();
+    await clusterListPage.isRegisterClusterUrl();
+    await clusterListPage.isRegisterClusterScreen();
+  });
+
+  test('shows an error when cluster ID is not valid', async ({ registerClusterPage }) => {
+    await registerClusterPage.clusterIDInput().fill('not really a uuid');
+    await registerClusterPage.clusterIDInput().blur();
+    await expect(registerClusterPage.clusterIDError()).toBeVisible();
+    await expect(registerClusterPage.clusterIDError()).toContainText(
+      "Cluster ID 'not really a uuid' is not a valid UUID.",
+    );
+    await registerClusterPage.clusterIDInput().clear();
+    await expect(registerClusterPage.clusterIDError()).toBeVisible();
+    await expect(registerClusterPage.clusterIDError()).toContainText('Cluster ID is required.');
+  });
+
+  test('shows error when display name is not valid', async ({ registerClusterPage }) => {
+    await registerClusterPage.displayNameInput().fill('a'.repeat(70));
+    await registerClusterPage.displayNameInput().blur();
+    await expect(registerClusterPage.displayNameError()).toBeVisible();
+    await expect(registerClusterPage.displayNameError()).toContainText(
+      'Cluster display name may not exceed 63 characters.',
+    );
+  });
+
+  test('shows error when URL is not valid', async ({ registerClusterPage }) => {
+    await registerClusterPage.clusterURLInput().fill('asdf');
+    await registerClusterPage.clusterURLInput().blur();
+    await expect(registerClusterPage.clusterURLError()).toBeVisible();
+    await expect(registerClusterPage.clusterURLError()).toContainText(
+      'The URL should include the scheme prefix (http://, https://)',
+    );
+    await registerClusterPage.clusterURLInput().clear();
+    await registerClusterPage.clusterURLInput().fill('https://uwu');
+    await registerClusterPage.clusterURLInput().blur();
+    await expect(registerClusterPage.clusterURLError()).toBeVisible();
+    await expect(registerClusterPage.clusterURLError()).toContainText('Invalid URL');
+  });
+
+  test('redirects to cluster list when clicking cancel', async ({
+    clusterListPage,
+    registerClusterPage,
+  }) => {
+    await registerClusterPage.cancelButton().click();
+    await clusterListPage.isClusterListScreen();
+  });
+
+  test('creates a new cluster and redirects to its details page', async ({
+    clusterListPage,
+    registerClusterPage,
+    clusterDetailsPage,
+  }) => {
+    await clusterListPage.registerCluster().click();
+    await clusterListPage.isRegisterClusterUrl();
+    await registerClusterPage.clusterIDInput().fill(clusterID);
+    await registerClusterPage.clusterIDInput().blur();
+    await expect(registerClusterPage.clusterIDError()).not.toBeVisible();
```

Comment:
Nitpick: In this and other cases in this file I see you are using `not.toBeVisible()` while the cypress equivalent checks for the existence of the element `should('not.exist')`. There's a difference, playwright has `not.toBeAttached()` for this purpose.